packages/hash/frontend/src/pages/entity-editor.page.tsx

```diff
@@ -57,9 +64,16 @@ const ExampleUsage = ({ ownedById }: { ownedById: string }) => {
   const propertyTypeDefinitions = useMemo(
     () =>
       entityTypeRootedSubgraph && propertyTypeIds
-        ? entityTypeRootedSubgraph.referencedPropertyTypes.filter(
-            ({ propertyTypeId }) => propertyTypeIds.includes(propertyTypeId),
-          )
+        ? propertyTypeIds.map((propertyTypeId) => {
+            const vertex = entityTypeRootedSubgraph.vertices[propertyTypeId];
+
+            /** @todo - what to do here if the property type isn't in the subgraph, make another call? */
```

Comment:
This is a wider question I have and am unsure of. It maybe can be solved when we also allow the depths to be configured through the BP functions?

cc: @CiaranMn for thoughts on BP ergonomics

Reply:
I expect in practice that most people are going to interact with types via queries that resolve all property, data, and link types referred to by a type. We should make that easy.

If they choose not to they can make another call to get missing things.

Reply:
I think for the `entity type -> property type -> data type` tree, having max depth parameters (255) is probably reasonable for most queries. I'm not sure about entity type -> entity type though. (Constraints on links for example), that could blow up the query very fast.

And it gets a lot less clear to me when we're dealing with knowledge graph stuff, not just ontology. For queries like `getEntity` or `aggregateEntities` do we really want to return the full `entity type -> property type -> data type` tree?

> It maybe can be solved when we also allow the depths to be configured through the BP functions?

Hence me thinking that. But it's probably good thinking about the patterns whereby we're going to need to make additional lookups (or account for it). Do we want to just assume that the full thing will fit in 255 (pretty safe bet for types in my eyes) and fatally error if it's not there? Or do we want to have fallback logic for follow-up queries on every single element kind, any time we look for something in the subgraph.

Reply:
> We should not pollute the 'get thing from subgraph' function with follow-up query logic though

100% on the same page there, otherwise you'd have to pass in all the hooks, it'd be a massive pain.

We can leave it as is then, I just have a **mild** concern we won't think about the edge-cases where we will want that until we discover a bug in prod